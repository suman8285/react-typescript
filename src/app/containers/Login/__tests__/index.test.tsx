import React from 'react';
import { Store } from '@reduxjs/toolkit';
import { fireEvent, screen, render, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { Login } from '../index';

interface IHistory {
  push: (url: string) => void;
}

interface ILocation {
  state: {
    [key: string]: string;
  };
}

const mockCallBack = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as object),
  useHistory: (): IHistory => ({
    push: mockCallBack,
  }),
  useLocation: (): ILocation => ({
    state: {
      from: '',
    },
  }),
}));

const store: Store = configureAppStore();

describe('Login Page', (): void => {
  it('Matching the snapshot', (): void => {
    const { asFragment } = render(
      <Provider store={store}>
        <Login storage={window.localStorage} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Login button should be defined', (): void => {
    render(
      <Provider store={store}>
        <Login storage={window.localStorage} />
      </Provider>,
    );
    const loginButton = screen.getAllByRole('button')[0];
    expect(loginButton).toBeDefined();
  });

  describe('Login page state', (): void => {
    beforeEach((): void => {
      render(
        <Provider store={store}>
          <Login storage={window.localStorage} />
        </Provider>,
      );
    });

    it('Should change the username-state on username input', (): void => {
      const inputUsername = screen.getAllByRole('textbox')[0];
      expect(inputUsername).toHaveValue('');
      fireEvent.change(inputUsername, { target: { value: 'chuck' } });
      expect(inputUsername).toHaveValue('chuck');
    });

    it('Should trigger the doLogin function on Login', async () => {
      const { asFragment } = render(
        <Provider store={store}>
          <Login storage={window.localStorage} />
        </Provider>,
      );

      const loginButton = screen.getAllByRole('button')[0];
      fireEvent.click(loginButton);

      await waitFor(() =>
        expect(store.getState().auth.initialized).toBeTruthy(),
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
