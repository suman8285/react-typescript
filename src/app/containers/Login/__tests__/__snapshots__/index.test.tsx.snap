// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Login page state Should trigger the doLogin function on Login 1`] = `
<DocumentFragment>
  .c1 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  box-sizing: border-box;
}

.c4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-image: linear-gradient(258deg,#93dc6d 5%,#1f9cc3);
  background-repeat: no-repeat;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50%;
  color: white;
  padding-left: 0px;
  padding-right: 0px;
}

.c3 {
  font-size: 52px;
  font-weight: 100;
  text-align: center;
  margin: 0;
  padding-top: 50px;
  padding-bottom: 10px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  background: transparent;
  padding: 20px 0;
  width: 450px;
}

.c6 {
  width: 100%;
  height: 60px;
  padding: 0 20px;
  font-size: 14px;
  border: 1px solid;
  margin-bottom: 10px;
}

.c6:focus-within {
  box-shadow: 0 0 15px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border: 0;
  background-color: green;
  border: 1px solid transparent;
  color: white;
  padding: 20px 0;
  font-size: 14px;
}

.c7:hover {
  cursor: pointer;
  background: green;
  border: 1px solid green;
}

.c7:focus {
  background: green;
  border: 1px solid green;
}

@media only screen and (min-width:1rem) {
  .c1 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:48rem) {
  .c1 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:64rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:90rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:120rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:1rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:48rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:64rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:90rem) {
  .c1 {
    width: 90rem;
  }
}

@media only screen and (min-width:120rem) {
  .c1 {
    width: 90rem;
  }
}

@media only screen and (min-width:1rem) {
  .c4 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:48rem) {
  .c4 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:64rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:90rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:120rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <p
        class="c3"
      >
        Sign In
      </p>
      <div
        class="c4 c5"
      >
        <input
          class="c6"
          id="username"
          placeholder="username"
          type="text"
          value=""
        />
        <input
          class="c6"
          id="password"
          placeholder="password"
          type="password"
          value=""
        />
        <button
          class="c7"
        >
          LOGIN
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login Page Matching the snapshot 1`] = `
<DocumentFragment>
  .c1 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  box-sizing: border-box;
}

.c4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-image: linear-gradient(258deg,#93dc6d 5%,#1f9cc3);
  background-repeat: no-repeat;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50%;
  color: white;
  padding-left: 0px;
  padding-right: 0px;
}

.c3 {
  font-size: 52px;
  font-weight: 100;
  text-align: center;
  margin: 0;
  padding-top: 50px;
  padding-bottom: 10px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  background: transparent;
  padding: 20px 0;
  width: 450px;
}

.c6 {
  width: 100%;
  height: 60px;
  padding: 0 20px;
  font-size: 14px;
  border: 1px solid;
  margin-bottom: 10px;
}

.c6:focus-within {
  box-shadow: 0 0 15px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border: 0;
  background-color: green;
  border: 1px solid transparent;
  color: white;
  padding: 20px 0;
  font-size: 14px;
}

.c7:hover {
  cursor: pointer;
  background: green;
  border: 1px solid green;
}

.c7:focus {
  background: green;
  border: 1px solid green;
}

@media only screen and (min-width:1rem) {
  .c1 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:48rem) {
  .c1 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:64rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:90rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:120rem) {
  .c1 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:1rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:48rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:64rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:90rem) {
  .c1 {
    width: 90rem;
  }
}

@media only screen and (min-width:120rem) {
  .c1 {
    width: 90rem;
  }
}

@media only screen and (min-width:1rem) {
  .c4 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:48rem) {
  .c4 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:64rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:90rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:120rem) {
  .c4 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <p
        class="c3"
      >
        Sign In
      </p>
      <div
        class="c4 c5"
      >
        <input
          class="c6"
          id="username"
          placeholder="username"
          type="text"
          value=""
        />
        <input
          class="c6"
          id="password"
          placeholder="password"
          type="password"
          value=""
        />
        <button
          class="c7"
        >
          LOGIN
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
